accepted/2020/reproducible-builds.md

```diff
@@ -0,0 +1,238 @@
+# Reproducible Build Validation
+
+**PM** [Claire Novotny](https://github.com/clairernovotny) |
+**Dev** [Jared Parsons](https://github.com/jaredpar)
+
+## Overview
+
+.NET's ecosystem uses binaries as the primary exchange mechanism. This solves various problems, especially shielding the consumer from having to replicate the build environment. It's also what enables the multi-language ecosystem. All in all, binaries have worked well for .NET, especially because the underlying format is rich and self-describing.
+
+.NET uses Authenticode signing for binaries and packages. The .NET Foundation has made it easier for member projects to get a code signing certificate and actively pushes its projects to do so. On top, the official NuGet gallery signs packages to indicate they came from nuget.org.
+
+However, one largely unsolved problem is the link between source code and binaries. While a significant portion of packages on nuget.org are open source and often link to a commit in a GitHub repo, there is no way to ensure that the source code at that commit actually matches the binaries submitted to nuget.org. In other words, just because a developer can trust the source code doesn’t mean they can trust the binaries came from that source. And while nuget.org scans packages for virus and malware, there is no way to detect all possible malware. For all they know, someone, or something, [injected a bitcoin miner](https://www.theregister.com/2018/11/26/npm_repo_bitcoin_stealer/).
+
+Also, should package consumers find a zero-day exploit, they usually have no way to patch the binaries themselves. They must find the corresponding repo, get the necessary dependencies for building it, somehow find the correct version of the code, patch it, and then rebuild. Not an easy task under ideal circumstances, much less viable when time is of the essence.
+
+To address these problems, we will create a set of tooling that for each managed binary file in a NuGet package, will be able to enumerate all of the input sources used to create it, validate they are accessible and validate that the resulting binary matches was produced using those sources files. In doing so, we don't need to trust the original build environment, as we're able to independently create the managed file from the input sources.
+
+## Scenarios and User Experience
+
+### Customer Cohorts
+
+- Package consumers who wish to manually check results on their own.
+- Package authors who wish to verify their package is reproducible.
+- Package hosts, such as NuGet.org and Terrapin, who wish to validate packages on upload and present the status as part of overall package/project health.
```

Comment:
One cohort not mentioned here, but was mentioned in introduction, is cohort who wants to patch a binary.

Reply:
In more recent discussions that cohort was essentially moved out of the primary scenarios. The focus at this point was more on provenance validation vs. patching. I agree though that's a scenario, in addition to several others, that we will likely expand into in the future.